### ISSUE TITLE
sqlalchemy use problem

### ISSUE THREAD

**Issue by 65tech** [mongkok/fastapi-debug-toolbar#12]
show "No SQL queries were recorded during this request."

**Comment by armando-herastang** [comment]
Also having this issue! Haven't look into it further. But no queries are shown

**Comment by Satchitananda** [comment]
Same here

**Comment by florimondmanca** [comment]
In my case, I found that this is because `fastapi-debug-toolbar` expects a `db: Session = Depends(...)` annotation on the route.

But:

* I don't use this style, as I use a different DI system than FastAPI's (I use [punq](https://pypi.org/project/punq/))
* I use SQLAlchemy async support, so the session would be an `AsyncSession` anyway.

To get things working, I had to subclass the default panel and register the SQLAlchemy engine myself:

```python
from debug_toolbar.panels.sqlalchemy import SQLAlchemyPanel as Base
from fastapi import Request, Response
from fastapi.encoders import jsonable_encoder

from server.config.di import resolve
from server.infrastructure.database import Database


class SQLAlchemyPanel(Base):
    async def process_request(self, request: Request) -> Response:
        db = resolve(Database)
        engine = db.engine.sync_engine

        self.register(engine)
        try:
            return await super().process_request(request)
        finally:
            self.unregister(engine)
```

(My actual implementation also includes the JSON dump fix from #15.)

Then use this by passing its Python module path to `panels=...`.

Then I am able to see queries in the SQLAlchemy section of debug toolbar.

Hope this helps!

**Comment by mongkok** [comment]
Hey @65tech ,
A new release was created related with this, see [v0.3.0](https://github.com/mongkok/fastapi-debug-toolbar/releases/tag/0.3.0).

If you are using FastAPI dependencies add `SQLAlchemyPanel` to the panel list, the panel supports now fastapi>=0.74.0.
If you don't use dependencies then create a new class that inherits from `SQLAlchemyPanel`, override the `add_engines` method and add the class path to your panel list, see #21:

```py
from debug_toolbar.panels.sqlalchemy import SQLAlchemyPanel as BasePanel
from sqlalchemy import create_engine

engine = create_engine("sqlite://...", connect_args={"check_same_thread": False})


class SQLAlchemyPanel(BasePanel):
    async def add_engines(self, request: Request):
        self.engines.add(engine)
```

Thanks @florimondmanca for your help on this.
